test/org/sagebionetworks/bridge/validators/StudyValidatorTest.java

```diff
@@ -1,95 +1,165 @@
-        DynamoStudy study = new DynamoStudy();
         study.getUserProfileAttributes().add("username");
         Validate.entityThrowingException(StudyValidator.INSTANCE, study);
     }
     
+    @Test(expected = InvalidEntityException.class)
+    public void requiresMissingSupportEmail() {
+        study.setSupportEmail(null);
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+    
+    @Test(expected = InvalidEntityException.class)
+    public void requiresMissingConsentNotificationEmail() {
+        study.setConsentNotificationEmail(null);
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+    
+    @Test(expected = InvalidEntityException.class)
+    public void requiresValidSupportEmail() {
+        study.setSupportEmail("asdf");
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+    
+    @Test(expected = InvalidEntityException.class)
+    public void requiresValidConsentNotificationEmail() {
+        study.setConsentNotificationEmail("aaa@aaa");
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+    
+    @Test(expected = InvalidEntityException.class)
+    public void requiresPasswordPolicy() {
+        study.setPasswordPolicy(null);
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+    
+    @Test(expected = InvalidEntityException.class)
+    public void requiresVerifyEmailTemplate() {
+        study.setVerifyEmailTemplate(null);
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+
+    @Test(expected = InvalidEntityException.class)
+    public void requiresVerifyEmailTemplateWithSubject() {
+        study.setVerifyEmailTemplate(new EmailTemplate("  ", "body", MimeType.HTML));
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+
+    @Test(expected = InvalidEntityException.class)
+    public void requiresVerifyEmailTemplateWithBody() {
+        study.setVerifyEmailTemplate(new EmailTemplate("subject", null, MimeType.HTML));
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+
+    @Test(expected = InvalidEntityException.class)
+    public void requiresResetPasswordTemplate() {
+        study.setResetPasswordTemplate(null);
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+
+    @Test(expected = InvalidEntityException.class)
+    public void requiresResetPasswordTemplateWithSubject() {
+        study.setResetPasswordTemplate(new EmailTemplate("  ", "body", MimeType.TEXT));
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
+
+    @Test(expected = InvalidEntityException.class)
+    public void requiresResetPasswordTemplateWithBody() {
+        study.setResetPasswordTemplate(new EmailTemplate("subject", null, MimeType.TEXT));
+        Validate.entityThrowingException(StudyValidator.INSTANCE, study);
+    }
```

Comment:
StudyValidator.validateTemplate() checks to make sure the body contains the ${url} template.